src/main/java/org/cactoos/iterable/Synced.java

```diff
@@ -73,15 +77,15 @@ public Synced(final Iterable<X> iterable) {
      * @param lck The lock to synchronize with.
      * @param iterable The iterable synchronize access to.
      */
-    public Synced(final Object lck, final Iterable<X> iterable) {
+    public Synced(final Object lck, final Iterable<? extends X> iterable) {
         this.origin = iterable;
         this.lock = lck;
     }
 
     @Override
     public Iterator<X> iterator() {
         synchronized (this.lock) {
-            return this.origin.iterator();
+            return new Mapped<>(x -> x, this.origin.iterator());
```

Comment:
@andreoss same here about cast vs mapped, and also about the todo

Reply:
@victornoel Replaced with casting.